internal/vault/client_factory_test.go

```diff
@@ -174,3 +174,143 @@ func Test_cachingClientFactory_clientLocks(t *testing.T) {
 		})
 	}
 }
+
+func Test_cachingClientFactory_callClientCallbacks(t *testing.T) {
+	ctx := context.Background()
+	type callbackResult struct {
+		called bool
+		done   bool
+	}
+	tests := []struct {
+		name       string
+		c          Client
+		onMask     ClientCallbackOn
+		cbOn       ClientCallbackOn
+		cbFn       func(t *testing.T) (ClientCallback, *callbackResult)
+		wait       bool
+		wantCalled bool
+	}{
+		{
+			name:   "single-on-lifetime-watcher-done",
+			c:      &defaultClient{},
+			onMask: ClientCallbackOnLifetimeWatcherDone,
+			cbOn:   ClientCallbackOnLifetimeWatcherDone,
+			cbFn: func(t *testing.T) (ClientCallback, *callbackResult) {
+				result := &callbackResult{}
+				return func(ctx context.Context, c Client) {
+					result.called = true
+					result.done = true
+				}, result
+			},
+			wantCalled: true,
+		},
+		{
+			name:   "single-on-lifetime-watcher-done",
+			c:      &defaultClient{},
+			onMask: ClientCallbackOnLifetimeWatcherDone,
+			cbOn:   ClientCallbackOnLifetimeWatcherDone,
+			cbFn: func(t *testing.T) (ClientCallback, *callbackResult) {
+				result := &callbackResult{}
+				return func(ctx context.Context, c Client) {
+					result.called = true
+					result.done = true
+				}, result
+			},
+			wantCalled: true,
+		},
```

Comment:
```suggestion
```
Looks like a duplicate of L193-206

Reply:
Good catch! Fixed in 21436f9fdbe383db166502b785efc76c8700d3ee